dashboard/test/models/census/census_summary_test.rb

```diff
@@ -36,4 +36,580 @@ class Census::CensusSummaryTest < ActiveSupport::TestCase
+          yes: 0,
+          no: 0,
+        },
+        not_teacher_or_admin: {
+          yes: 0,
+          no: 5,
+        },
+      }
+    }
+
+    teaches_cs = compute_teaches_cs args
+
+    assert_equal 'NO', teaches_cs, args[:audit]
+    explanation = args[:audit][:explanation]
+    validate_explanation explanation
+    explanation.select {|e| e[:label] == 'non_teacher_surveys'}.first.tap do |e|
+      assert e[:used], e
+      assert_equal({yes: 0, no: 5}, e[:value], e)
+    end
+  end
+
+  test "compute_teaches_cs with inconsistent teacher surveys gives correct results" do
+    args = empty_compute_teaches_cs_args
+    args[:submissions_summary] = {
+      consistency: {
+        teacher_or_admin: nil,
+        not_teacher_or_admin: nil,
+      },
+      has_inconsistent_surveys: {
+        teacher_or_admin: true,
+        not_teacher_or_admin: true,
+      },
+      counts: {
+        teacher_or_admin: {
+          yes: 5,
+          no: 1,
+        },
+        not_teacher_or_admin: {
+          yes: 5,
+          no: 5,
+        },
+      }
+    }
+
+    teaches_cs = compute_teaches_cs args
+
+    assert_equal 'MAYBE', teaches_cs, args[:audit]
+    explanation = args[:audit][:explanation]
+    validate_explanation explanation
+    explanation.select {|e| e[:label] == 'inconsistent_teacher_surveys'}.first.tap do |e|
+      assert e[:used], e
+      assert_equal({yes: 5, no: 1}, e[:value], e)
+    end
+    explanation.select {|e| e[:label] == 'non_teacher_surveys'}.first.tap do |e|
+      refute e[:used], e
+      assert_equal({yes: 5, no: 5}, e[:value], e)
+    end
+  end
+
+  test "compute_teaches_cs with last year yes gives correct results" do
+    validate_compute_teaches_cs_with_last_years_value('YES', 'HISTORICAL_YES', true)
+  end
+
+  test "compute_teaches_cs with last year no gives correct results" do
+    validate_compute_teaches_cs_with_last_years_value('NO', 'HISTORICAL_NO', true)
+  end
+
+  test "compute_teaches_cs with last year maybe gives correct results" do
+    validate_compute_teaches_cs_with_last_years_value('MAYBE', 'HISTORICAL_MAYBE', true)
+  end
+
+  test "compute_teaches_cs with last year historical yes gives correct results" do
+    validate_compute_teaches_cs_with_last_years_value('HISTORICAL_YES', nil, false)
+  end
+
+  test "compute_teaches_cs with last year historical no gives correct results" do
+    validate_compute_teaches_cs_with_last_years_value('HISTORICAL_NO', nil, false)
+  end
+
+  test "compute_teaches_cs with last year historical maybe gives correct results" do
+    validate_compute_teaches_cs_with_last_years_value('HISTORICAL_MAYBE', nil, false)
+  end
+
+  test "compute_teaches_cs with last year unknown gives correct results" do
+    validate_compute_teaches_cs_with_last_years_value('UNKNOWN', nil, false)
+  end
+
+  test "compute_teaches_cs with last year excluded gives correct results" do
+    validate_compute_teaches_cs_with_last_years_value('EXCLUDED', nil, false)
+  end
```

Comment:
These tests (plus the similar ones on [lines 513-519](https://github.com/code-dot-org/code-dot-org/pull/53972/files#diff-a14752a56427bc9310e7734fb75aed35684a55eb87349095735ba8da0930d6ecR513-R519)) are new, and the rest are from the [original tests](https://github.com/code-dot-org/code-dot-org/commit/e47561190f834976b596665bab0ade4717191af8#diff-a14752a56427bc9310e7734fb75aed35684a55eb87349095735ba8da0930d6ec).